File: patterns/hysteresis/rules-js/hysteresis.js
rules.JSRule({
  name: "Heizung mit Hysterese",
  triggers: [triggers.ItemStateChangeTrigger("MyTemp")],
  execute: (event) => {
    const temp = parseFloat(items.getItem("MyTemp").state);
    let newCommand = "STAY";

    if (temp < 68) {
      newCommand = "ON";
    } else if (temp >= 70) {
      newCommand = "OFF";
    }

    if (newCommand !== "STAY" && newCommand !== items.getItem("MyHeater").state) {
      items.getItem("MyHeater").sendCommand(newCommand);
    }
  }
});

File: patterns/hysteresis/rules-python/hysteresis.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger


@rule(triggers=[ItemStateChangeTrigger("MyTemp")])
class HeizungMitHysterese:
    def execute(self, module, input):
        temp = float(str(Registry.getItem("MyTemp").getState()))
        new_command = "STAY"

        if temp < 68:
            new_command = "ON"
        elif temp >= 70:
            new_command = "OFF"

        heater = Registry.getItem("MyHeater")
        if new_command != "STAY" and new_command != str(heater.getState()):
            heater.sendCommand(new_command)
            self.logger.info("Temp={} -> Heizung={}".format(temp, new_command))
        else:
            self.logger.info("Temp={} -> keine Aenderung (Hysterese-Bereich oder bereits im Zielzustand)".format(temp))
